### PR TITLE
Implement UniqueID on Darwin

### PR DESCRIPTION
**Pull request by cwurm**
Implements https://github.com/elastic/go-sysinfo/issues/29.

### CODE REVIEW

File: providers/darwin/host_darwin_amd64.go [hunk 1]
```diff
@@ -218,5 +224,23 @@ func (r *reader) time(h *host) {
 }
 
 func (r *reader) uniqueID(h *host) {
-	// TODO: call gethostuuid(uuid [16]byte, timespec)
```
[the remark this answers is not in the document]
Comment (cwurm) [reply providers/darwin/host_darwin_amd64.go]: done

File: providers/darwin/host_darwin_amd64.go [hunk 2]
```diff
@@ -218,5 +224,23 @@ func (r *reader) time(h *host) {
...
+	var uuidC C.uuid_t
+	const size = unsafe.Sizeof(uuidC)
+	var id [size]C.uchar
+	wait := C.struct_timespec{5, 0} // 5 seconds
+
+	ret, err := C.gethostuuid(&id[0], &wait)
+	if ret == 0 && err == nil {
+		bytes := C.GoBytes(unsafe.Pointer(&id[0]), C.int(size))
+		h.info.UniqueID = hex.EncodeToString(bytes)
```
[the remark this answers is not in the document]
Comment (cwurm) [reply providers/darwin/host_darwin_amd64.go]: It is. I've added `uuid_unparse_upper` to have it look exactly the same as it does on macOS.

File: providers/darwin/host_darwin_amd64.go [hunk 3]
```diff
@@ -218,5 +224,23 @@ func (r *reader) time(h *host) {
...
 
 func (r *reader) uniqueID(h *host) {
-	// TODO: call gethostuuid(uuid [16]byte, timespec)
+	var uuidC C.uuid_t
+	const size = unsafe.Sizeof(uuidC)
+	var id [size]C.uchar
+	wait := C.struct_timespec{5, 0} // 5 seconds
+
+	ret, err := C.gethostuuid(&id[0], &wait)
+	if ret == 0 && err == nil {
```
[the remark this answers is not in the document]
Comment (cwurm) [reply providers/darwin/host_darwin_amd64.go]: 👍

**Comment by cwurm** [comment]
Even though all three platforms (windows, linux, darwin) have UUIDs as their UniqueIDs they would each be displayed in a different way:

- Linux is currently all lowercase, no hyphens: `6f7be6fb33e6c67f057266415c084408`
- Windows is currently all lowercase, with hyphens: `fa4a0df6-5a59-4870-84a0-cde535805dfc`
- MacOS would be all uppercase, with hyphens: `155320EF-9DB6-5949-966B-F074A556AD32`

Would it make sense to unify?

File: providers/darwin/machineid_darwin_amd64.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+// #include <uuid/uuid.h>
+import "C"
+
+import (
+	"unsafe"
+
+	"github.com/pkg/errors"
+)
+
+func MachineID() (string, error) {
```
Comment (andrewkroh) [inline providers/darwin/machineid_darwin_amd64.go]: Can you add godocs for this that explain that it returns the hardware ID.

Comment (cwurm) [reply providers/darwin/machineid_darwin_amd64.go]: done

**Pull request merged**